Assets/Scripts/Buddy/BuddyShaper.cs

```diff
@@ -53,28 +60,46 @@ void Start()
 		}
 	}
 
-	void RandomizeBuddy()
+	void Update()
 	{
-		for( int i = 0; i < _numBlendShapes; i++ )
+		if( _debugRandomizer != _prevDebugRandomizer )
 		{
-			_skinnedMeshRend.SetBlendShapeWeight( i, 0f );
+			if( _debugRandomizer )
+			{
+				InvokeRepeating( "RandomizeBuddy", 0.3f, 0.3f );
+			}
+			else
+			{
+				CancelInvoke( "RandomizeBuddy");
+			}
+
+			_prevDebugRandomizer = _debugRandomizer;
 		}
```

Comment:
The alternative I thought of would be a coroutine that runs forever and checks if it should continue randomizing after every randomize/waitforseconds cycle. That seems just as bad to me and apparently Invoke is faster than a coroutine. I'm open to suggestions.